platforms/win32/vm/sqWin32Main.c

```diff
@@ -462,316 +467,310 @@ char *hwInfoString = "";
-      GetProcAddress(hUser, "EnumDisplayDevicesA");
-    ZeroMemory(&gDev, sizeof(gDev));
-    gDev.cb = sizeof(gDev);
-    if(pEnumDisplayDevices) pEnumDisplayDevices(NULL, 0, &gDev, 0);
-  }
-
-  { /* Figure out make and model from OEMINFO.ini */
-    char iniName[256];
-    char manufacturer[256];
-    char model[256];
-
-    GetSystemDirectory(iniName, 256);
-    strcat(iniName,"\\OEMINFO.INI");
-
-    GetPrivateProfileString("General", "Manufacturer", "Unknown", 
-			    manufacturer, 256, iniName);
-
-    GetPrivateProfileString("General", "Model", "Unknown", 
-			    model, 256, iniName);
-
-    sprintf(tmpString, 
-	    "Hardware information: \n"
-	    "\tManufacturer: %s\n"
-	    "\tModel: %s\n"
-	    "\tNumber of processors: %lu\n"
-	    "\tPage size: %lu\n"
-	    "\nMemory Information (upon launch):\n"
-	    "\tPhysical Memory Size: %" PRIuSQPTR " kbytes\n"
-	    "\tPhysical Memory Free: %" PRIuSQPTR " kbytes\n"
-	    "\tPage File Size: %" PRIuSQPTR " kbytes\n"
-	    "\tPage File Free: %" PRIuSQPTR " kbytes\n"
-	    "\tVirtual Memory Size: %" PRIuSQPTR " kbytes\n"
-	    "\tVirtual Memory Free: %" PRIuSQPTR " kbytes\n"
-	    "\tMemory Load: %lu percent\n",
-	    manufacturer, model,
-	    sysInfo.dwNumberOfProcessors,
-	    sysInfo.dwPageSize,
-	    memStat.dwTotalPhys / 1024,
-	    memStat.dwAvailPhys / 1024,
-	    memStat.dwTotalPageFile / 1024,
-	    memStat.dwAvailPageFile / 1024,
-	    memStat.dwTotalVirtual / 1024,
-	    memStat.dwAvailVirtual / 1024,
-	    memStat.dwMemoryLoad);
-  }
-
-  /* find more information about each processor */
-  for(proc=0; proc < sysInfo.dwNumberOfProcessors; proc++) {
-
-    char *tmp = tmpString + strlen(tmpString);
-
-    strcpy(keyName, "HARDWARE\\DESCRIPTION\\System\\CentralProcessor\\0");
-    keyName[strlen(keyName)-1] = 48+proc; /* 0, 1, 2 etc. */
-
-    ok = RegOpenKey(HKEY_LOCAL_MACHINE, keyName, &hk);
-    if(ok == 0) {
-      char nameString[256];
-      char identifier[256];
+    /* Figure out make and model from OEMINFO.ini */
+    TCHAR iniName[MAX_PATH];
+    TCHAR manufacturer[SYSTEM_INFO_BUF_SIZE];
+    TCHAR model[SYSTEM_INFO_BUF_SIZE];
+
+    GetSystemDirectory(iniName, MAX_PATH);
+    lstrcat(iniName, TEXT("\\OEMINFO.INI"));
+
+    GetPrivateProfileString(TEXT("General"),
+                            TEXT("Manufacturer"), TEXT("Unknown"),
+                            manufacturer, SYSTEM_INFO_BUF_SIZE, iniName);
+
+    GetPrivateProfileString(TEXT("General"),
+                            TEXT("Model"), TEXT("Unknown"),
+                            model, SYSTEM_INFO_BUF_SIZE, iniName);
+
+
+    _stprintf(tmpString,
+              TEXT("Hardware information: \n")
+              TEXT("\tManufacturer: %s\n")
+              TEXT("\tModel: %s\n")
+              TEXT("\tNumber of processors: %d\n")
+              TEXT("\tPage size: %d\n")
+              TEXT("\nMemory Information (upon launch):\n")
+              TEXT("\tPhysical Memory Size: %d kbytes\n")
```

Comment:
Yes, ugly, but correct 
C99-like %" PRIuSQPTR " is ugly per se
and there are several instances already of TEXT(PRI...)